fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -242,13 +247,17 @@ public void setMaintainHistory(boolean maintainHistory) {
     }
 
     @Nonnull
-    protected RecordMetaData buildMetaData(RecordMetaDataProto.MetaData metaDataProto, boolean validate) {
+    protected RecordMetaDataBuilder createMetaDataBuilder(RecordMetaDataProto.MetaData metaDataProto) {
         RecordMetaDataBuilder builder = RecordMetaData.newBuilder().addDependencies(dependencies);
         if (localFileDescriptor != null) {
             builder.setLocalFileDescriptor(localFileDescriptor);
```

Comment:
I the user has initialized the `FDBMetaDataStore` with a `localFileDescriptor`, won't this mean that the `updateRecords` method fails (somewhat mysteriously)? I think we would want it to create the meta-data without the local file descriptor for the `updateRecords` call and then (maybe) add it back in in `saveAndSetCurrent`.

Reply:
I actually think we should fail `updateRecords` if `localFileDescriptor` is set. I.e., `updateRecords` is not supported when `localFileDescriptor` is used and there shouldn't be any exception here either. Maybe that means we should not support `updateRecords` in `FDBMetaDataStore` at all. What do you think?